eslint.config.js

```diff
@@ -62,7 +62,7 @@ export default [
 			// These off/configured-differently-by-default rules fit well for us
 			'@typescript-eslint/switch-exhaustiveness-check': 'error',
 			'@typescript-eslint/no-unused-vars': [
-				'warn',
+				'error',
```

Comment:
We've historically been against this as some of us didn't like the experience of having errors while working, so I guess we should probably discuss it a bit

Reply:
Given what the PR has caught if we had a stricter lint, I think this is a nice change to me. It makes the code more readable too, which improves the experience for others working on the code.